kasa/httpclient.py

```diff
@@ -1,34 +1,37 @@
 """Module for HttpClientSession class."""
-import logging
-from typing import Any, Dict, Optional, Tuple, Type, Union
+from typing import Any, Dict, Optional, Tuple, Union
 
-import httpx
+import aiohttp
 
 from .deviceconfig import DeviceConfig
 from .exceptions import ConnectionException, SmartDeviceException, TimeoutException
+from .json import loads as json_loads
 
-logging.getLogger("httpx").propagate = False
 
-InnerHttpType = Type[httpx.AsyncClient]
+def get_cookie_jar() -> aiohttp.CookieJar:
+    """Return a new cookie jar with the correct options for device communication."""
+    return aiohttp.CookieJar(unsafe=True, quote_cookie=False)
 
 
 class HttpClient:
     """HttpClient Class."""
 
     def __init__(self, config: DeviceConfig) -> None:
         self._config = config
-        self._client: httpx.AsyncClient = None
+        self._client: aiohttp.ClientSession = None
+        self._jar = aiohttp.CookieJar(unsafe=True, quote_cookie=False)
+        self._last_url = f"http://{self._config.host}/"
 
     @property
-    def client(self) -> httpx.AsyncClient:
+    def client(self) -> aiohttp.ClientSession:
         """Return the underlying http client."""
         if self._config.http_client and issubclass(
-            self._config.http_client.__class__, httpx.AsyncClient
+            self._config.http_client.__class__, aiohttp.ClientSession
```

Comment:
It'd be better to raise an exception instead of silently falling back to the default  if an unexpected client is passed in.